HumanDynamicsEstimationLibrary/algorithms/src/DynamicalInverseKinematics.cpp

```diff
@@ -0,0 +1,1378 @@
+                                    const double orientationFeedbackGain,
+                                    const double angularVelocityFeedforwardGain,
+                                    const double angularVelocityWeight)
+{
+    return pImpl->addTarget(DynamicalInverseKinematics::impl::InverseKinematicsTarget::OrientationAndVelocityTarget(
+        linkName, orientation, angularVelocity, orientationFeedbackGain, angularVelocityFeedforwardGain, angularVelocityWeight));
+}
+
+bool DynamicalInverseKinematics::addOrientationTarget(const std::string& linkName,
+                        const iDynTree::Rotation& orientation,
+                        const double orientationFeedbackGain,
+                        const double angularVelocityWeight)
+{
+    return pImpl->addTarget(DynamicalInverseKinematics::impl::InverseKinematicsTarget::OrientationTarget(
+        linkName, orientation, orientationFeedbackGain, angularVelocityWeight));
+}
+
+
+bool DynamicalInverseKinematics::setJointConfiguration(const std::string& jointName,
+                                                       const double jointConfiguration)
+{
+    if (!pImpl->m_isModelLoaded)
+        return false;
+    
+    iDynTree::JointIndex jointIndex = pImpl->m_dynamics.model().getJointIndex(jointName);
+    if (jointIndex == iDynTree::JOINT_INVALID_INDEX)
+        return false;
+    pImpl->m_state.s(jointIndex) = jointConfiguration;
+    pImpl->updateConfiguration();
+    return true;
+}
+
+bool DynamicalInverseKinematics::setJointsConfiguration(const iDynTree::VectorDynSize& jointsConfiguration)
+{
+    if (!pImpl->m_isModelLoaded)
+        return false;
+    
+    if (pImpl->m_state.s.size() == jointsConfiguration.size()) {
+        pImpl->m_state.s = jointsConfiguration;
+        pImpl->updateConfiguration();
+        return true;
+    }
+    else {
+        return false;
+    }
+}
+
+bool DynamicalInverseKinematics::setBasePose(const iDynTree::Transform& baseTransform)
+{
+    if (!pImpl->m_isModelLoaded)
+        return false;
+    
+    pImpl->m_state.W_p_B = baseTransform.getPosition();
+    pImpl->m_state.W_R_B = baseTransform.getRotation();
+    pImpl->updateConfiguration();
+    return true;
+}
+
+bool DynamicalInverseKinematics::setBasePose(const iDynTree::Vector3& basePosition,
+                                            const iDynTree::Rotation& baseRotation)
+{
+    if (!pImpl->m_isModelLoaded)
+        return false;
+    
+    pImpl->m_state.W_p_B = basePosition;
+    pImpl->m_state.W_R_B = baseRotation;
+    pImpl->updateConfiguration();
+    return true;
+}
+
+bool DynamicalInverseKinematics::setConfiguration(const iDynTree::Transform& baseTransform,
+                                                 const iDynTree::VectorDynSize& jointsConfiguration)
+{
+    if (!pImpl->m_isModelLoaded)
+        return false;
+    
+    if (setJointsConfiguration(jointsConfiguration) && setBasePose(baseTransform)) {
+        pImpl->updateConfiguration();
```

Comment:
`pImpl->updateConfiguration()` is already called in the `setJointsConfiguration(jointsConfiguration) & setBasePose(baseTransform)`. I am not sure if it is needed here as well, or it is a duplication!

Reply:
fixed in https://github.com/robotology/human-dynamics-estimation/pull/265/commits/464177c5d794cb35c8739bdf76a23de82ed1cc2d